daffodil-core/src/main/scala/org/apache/daffodil/dsom/RawCommonRuntimeValuedPropertiesMixin.scala

```diff
@@ -64,8 +64,6 @@ trait RawSequenceRuntimeValuedPropertiesMixin
 
 trait RawLayeringRuntimeValuedPropertiesMixin
   extends PropertyMixin {
-  protected final lazy val optionLayerTransformRaw = findPropertyOption("layerTransform", expressionAllowed = true)
-  protected final lazy val layerTransformRaw = requireProperty(optionLayerTransformRaw)
   protected final lazy val optionLayerEncodingRaw = findPropertyOption("layerEncoding", expressionAllowed = true)
```

Comment:
The general principle for DFDL was all properties that are needed are required to be expressed. However, now that DFDL v1.0 is finished, new properties added would need a control over when to force that behavior, and when to tolerate them being missing. 

The dfdlx:layerEncoding property is only required if the layer transform needs an encoding, and it furthermore only needs it if the xs:sequence, on which the layer is expressed either doesn't have the property at all (perhaps because it doesn't need it), or the layer transform requires a different encoding property than the xs:sequence. 

I expect a primary use case to be a textual data document in UTF-8, containing say, a base64-encoded gzipped region, which we know can only contain ascii characters. The dfdlx:layerEncoding can specify ascii, which more accurately reflects the format constraint on that region.  That's the notion anyway.

Reply:
> The dfdlx:layerEncoding property is only required if the layer transform needs a

I think this is the thing that feels odd to me. We have a DFDL extension property that is specific to text-related layers, some of which won't even use it because they assume some encoding. For something that will soon allow pluggable transforms, it seems we also need some way to have pluggable properties specific to those transforms. To me encoding feels like it wants to be that kind of property property because it only applies to certain transforms. Maybe a key/value pair similar to dfdlx:runtimeProperties. The other option is to just add new dfdlx:layer* properties everytime a layer needs a property, but that doesn't scale well, and also makes thigns diffcult for pluggable layers we don't/can't know about.